### DISCUSSION TITLE
关于异步的使用问题？

### DISCUSSION THREAD

**Discussion by SamaHddd** [BYJRK/DotNet-Discussions#272] [C# 技术交流]
请问我后台有一个需要长时间读取数据的任务，并且实时根据数据更新UI。这时候用Task合适嘛？

**Comment by BYJRK** [comment]
你或许可以考虑使用异步流：https://learn.microsoft.com/en-us/dotnet/csharp/asynchronous-programming/generate-consume-asynchronous-stream

**Comment by SlimeNull** [comment]
你的数据源提供异步方法了吗，如果提供了，那就用。如果没有，那你只能用线程。

**Comment by freeman888** [comment]
个人感觉task提供的线程池还是比较适合相对短时间的任务，那种在一定时间内可以完成的任务。长时间轮询来获取数据的任务用线程还是比较好，但是和主线程的通信可以考虑用TaskCompletionSource，可以在形式上使用async/await那一套。

**Comment by CodingOctocat** [comment]
突然想起之前 star 过的一个项目：https://www.hangfire.io/

![image](https://github.com/user-attachments/assets/c8df436c-bd08-402c-bc79-e973888ddba9)

![image](https://github.com/user-attachments/assets/b1341a5e-807e-4c05-ab9c-4a1fda49f928)
